src/driver/aurora-mysql/AuroraMysqlDriver.ts

```diff
@@ -633,6 +633,9 @@ export class AuroraMysqlDriver implements Driver {
         ) {
             // convert to number if that exists in possible enum options
             value = parseInt(value)
+        } else if (columnMetadata.type === Number) {
+            // convert to number if number
+            value = !isNaN(+value) ? parseInt(value) : value
```

Comment:
can you please explain how this change related to the proposed functionality.

Reply:
This change was made in all of the rdb drivers because typeorm was not casting the data types automatically based on my models type definition (specifically for numbers) which hindered my development for this feature.

IE: When I pulled the data from the DB, I was expecting my data type to be a number because that is what I defined in my virtual column decorator's metadata but the returned result was a string.

If you are specifically asking about this driver, I can remove it for this one but did update all rdb drivers.

Reply:
@pleerock @kerhong - The above is my explanation as to why this was needed.

If number is specified without the column type, we assume "int" as Number unless specified in the @Column() decorator - if this is not the case and it should be assumed as a float, we could parseFloat() here but the specification @pleerock mentioned below is that it should be assumed as an integer